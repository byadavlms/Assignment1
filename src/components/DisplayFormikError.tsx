import { ErrorMessage } from "formik";

const DisplayFormikError = ({ name }: { name: string }) => (
    <ErrorMessage name={name}>
        {(msg) => (
            <span style={{
                display: 'block',
                color: 'red',
                marginLeft: '130px'
            }}>
                {msg}
            </span>
        )}
    </ErrorMessage>
);

export default DisplayFormikError;
