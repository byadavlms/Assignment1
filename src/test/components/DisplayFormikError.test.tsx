import { shallow } from 'enzyme';
import DisplayFormikError from '../../components/DisplayFormikError';

jest.mock('formik'); // formik package is auto mocked

describe('DisplayFormikError', () => {
    it('renders correctly', () => {
        const component = shallow(<DisplayFormikError name='fieldName' />);

        expect(component).toMatchSnapshot();
        // On the first run of this test, Jest will generate a snapshot file automatically.
    });

    it('should render and recieve name as props', () => {
        const component = shallow(<DisplayFormikError name='fieldName' />);

        expect(component.props().name).toEqual('fieldName');
    });
});
