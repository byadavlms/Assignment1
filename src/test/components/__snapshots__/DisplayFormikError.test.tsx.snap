// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayFormikError renders correctly 1`] = `
<FormikConnect(ErrorMessageImpl)
  name="fieldName"
>
  <Component />
</FormikConnect(ErrorMessageImpl)>
`;

exports[`DisplayFormikError renders correctly 2`] = `
<FormikConnect(ErrorMessageImpl)
  name="fieldName"
>
  <Component />
</FormikConnect(ErrorMessageImpl)>
`;

exports[`DisplayFormikError should mount and recieve name as props 1`] = `
<FormikConnect(ErrorMessageImpl)
  name="fieldName"
>
  <Component />
</FormikConnect(ErrorMessageImpl)>
`;
